### ISSUE TITLE
Error when branch name contains a forward slash

### ISSUE THREAD

**Issue by xtqqczze** [MihaZupan/runtime-utils#3]
> @MihaZupan runtime-utils doesn't handle a branch containing a forward slash properly, i.e.:
> 
> https://github.com/xtqqczze/dotnet-runtime/tree/draft/VectorGetArrayDataReference
> 
> ```log
> [00:00:34] [Setup runtime] + git remote add combineWith1 https://github.com/xtqqczze/dotnet-runtime
> [00:00:34] [Setup runtime] + git fetch combineWith1 draft
> [00:00:34] [Setup runtime] fatal: couldn't find remote ref draft
> ```

_Originally posted by @xtqqczze in https://github.com/MihuBot/runtime-utils/issues/491#issuecomment-2218882702_

**Comment by MihaZupan** [comment]
This isn't an issue if you start the bot by mentioning it on a PR since that's taking the branch from GH's API:
https://github.com/MihaZupan/MihuBot/blob/ab1e739fac5d3a8f018f4e967b7a3272ae8e475b/MihuBot/MihuBot/RuntimeUtils/JobBase.cs#L97

If you start it with a custom branch from the Web UI, we'd try to parse it from the URL instead.
https://github.com/MihaZupan/MihuBot/blob/ab1e739fac5d3a8f018f4e967b7a3272ae8e475b/MihuBot/MihuBot/Pages/RuntimeUtils.razor#L347-L355
I don't think there's a way to handle slashes without calling into GH's API to get a list of all branches.
Looks like that part's also missing allowing the `.` in the Regex pattern.

**Comment by xtqqczze** [comment]
Yeah, I also came to the conclusion it's not possible to use regex to handle forward slashes while still handling, i.e.:
https://github.com/MihaZupan/performance/blob/regex/.gitignore#L5
